contracts/optimism/connectors/aave/aave-rewards/v3/main.sol

```diff
@@ -0,0 +1,78 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+/**
+ * @title Aave v3 Rewards.
+ * @dev Claim Aave v3 rewards.
+ */
+
+import { TokenInterface } from "../../../../common/interfaces.sol";
+import { Stores } from "../../../../common/stores.sol";
+import { Helpers } from "./helpers.sol";
+import { Events } from "./events.sol";
+
+abstract contract IncentivesResolver is Helpers, Events {
+	/**
+	 * @dev Claim Pending Rewards.
+	 * @notice Claim Pending Rewards from Aave v3 incentives contract.
+	 * @param assets The list of assets supplied and borrowed.
+	 * @param amt The amount of reward to claim. (uint(-1) for max)
+	 * @param reward The address of reward token to claim.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of rewards claimed.
+	 */
+	function claim(
+		address[] calldata assets,
+		uint256 amt,
+		address reward,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(getId, amt);
+
+		require(assets.length > 0, "invalid-assets");
+
+		_amt = incentives.claimRewards(assets, _amt, address(this), reward);
+
+		TokenInterface weth = TokenInterface(wethAddr);
+		uint256 wethAmount = weth.balanceOf(address(this));
+		convertWethToEth(wethAmount > 0, weth, wethAmount);
```

Comment:
@Richa-iitr We only need to do this when `reward` is weth, right? 

@pradyuman-verma ?

Reply:
Yes